.github/workflows/release.yaml

```diff
@@ -65,12 +65,49 @@ jobs:
             echo "Tag '${VERSION}' does not exist."
           fi
 
-  build_images:
+  release_sparkctl:
     needs:
       - check-release
 
     runs-on: ubuntu-latest
 
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - linux
+          - darwin
+        arch:
+          - amd64
+          - arm64
```

Comment:
The matrix here as no impact on the os / arch use since we rely on ubuntu-latest (linux/amd64)

But we could rely on go native capability to build cross-platform binaries by setting `GOOS=${{ matrix.os }} GOARCH=${{ matrix.arch }}` as env var when running `make build-sparkctl`

I would not recommend using matrix in that specific case because It create new runner for each variation and does not reuse the go build cache meaning that time we have to wait a full build time

Reply:
> But we could rely on go native capability to build cross-platform binaries by setting `GOOS=${{ matrix.os }} GOARCH=${{ matrix.arch }}` as env var when running `make build-sparkctl`

Make sense. Have added them as job envs.

> I would also not recommend using matrix in that specific case because It create new runner for each variation and does not reuse the go build cache meaning that time we have to wait a full build time

It will truly create 4 different runners, but they will build sparkctl binary concurrently, so actually the whole waiting time is shorter.